sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/io/TextIO.java

```diff
@@ -340,6 +341,14 @@ private Bound(String name, String filepattern, Coder<T> coder, boolean validate,
       }
 
       @Override
+      public void populateDisplayData(DisplayData.Builder builder) {
+        builder.add("compressionType", compressionType.toString());
+        if (filepattern != null) {
+          builder.add("filePattern", filepattern);
+        }
```

Comment:
Same comment as from Avro-- the source isn't created until within `.apply()`. I'm not sure if it's worth it (or safe to) create a temporary Source just for display data.